fe/fe-core/src/main/java/org/apache/doris/analysis/UpdateStmt.java

```diff
@@ -170,7 +169,7 @@ private void analyzeSetExprs(Analyzer analyzer) throws AnalysisException {
                 throw new AnalysisException("The left side of the set expr must be the column name");
             }
             lhs.analyze(analyzer);
-            if (((SlotRef) lhs).getColumn().getAggregationType() != AggregateType.REPLACE) {
+            if (((SlotRef) lhs).getColumn().isKey()) {
```

Comment:
Could you add some details in this pr to explain the difference between this and the original one.

Reply:
AggregateType of value column is NONE in unique table with merge on write. In order to be compatible with these two scenarios, `isKey` is used to determine whether it is value column or key column.